components/mediation/data-publishers/org.wso2.micro.integrator.observability/src/main/java/org/wso2/micro/integrator/observability/handler/metrichandler/MetricHandler.java

```diff
@@ -0,0 +1,440 @@
+                        synCtx.getProperty(SynapseConstants.IS_INBOUND).toString().equals("true")) {
+                    String inboundEndpointName = synCtx.getProperty(SynapseConstants.INBOUND_ENDPOINT_NAME).
+                            toString();
+                    incrementInboundEndpointErrorCount(inboundEndpointName);
+                    stopTimers(synCtx.getProperty
+                            (MetricConstants.INBOUND_ENDPOINT_LATENCY_TIMER), synCtx);
+                } else {
+                    serviceInvokePort = getServiceInvokePort(synCtx);
+                    internalHttpApiPort = getInternalHTTPInboundEndpointPort(synCtx);
+
+                    if (null != synCtx.getProperty(RESTConstants.SYNAPSE_REST_API) &&
+                            (serviceInvokePort != internalHttpApiPort)) {
+                        String context = axis2MessageContext.getProperty(MetricConstants.TRANSPORT_IN_URL).
+                                toString();
+                        String apiInvocationUrl = axis2MessageContext.getProperty(MetricConstants.SERVICE_PREFIX).
+                                toString() + context.replaceFirst(DELIMITER, EMPTY);
+                        String apiName = getApiName(context, synCtx);
+
+                        incrementApiErrorCount(apiName, apiInvocationUrl);
+                        stopTimers(synCtx.getProperty(MetricConstants.API_LATENCY_TIMER), synCtx);
+                    }
+                }
+            }
+        }
+        synCtx.setProperty(SynapseConstants.IS_ERROR_COUNT_ALREADY_PROCESSED, true);
+        return true;
+    }
+
+    @Override
+    public boolean handleServerShutDown() {
+        metricReporterInstance.serverDown(host, port, javaHome, javaVersion);
+        return true;
+    }
+
+    @Override
+    public boolean handleArtifactDeployment(String artifactName, String artifactType, String startTime) {
+        metricReporterInstance.serviceUp(artifactName, artifactType);
+        return true;
+    }
+
+    @Override
+    public boolean handleArtifactUnDeployment(String artifactName, String artifactType, String startTime) {
+        metricReporterInstance.serviceDown(artifactName, artifactType);
+        return true;
+    }
+
+    /**
+     * Start the timers to observe request latency.
+     *
+     * @param synCtx           The Synapse Message Context
+     * @param serviceName      The proxy/api/inbound endpoint name
+     * @param serviceType      The service type (proxy/api/inbound endpoint)
+     * @param apiInvocationUrl The api invocation url
+     */
+    private void startTimers(MessageContext synCtx, String serviceName, String serviceType, String apiInvocationUrl) {
+        switch (serviceType) {
+            case SynapseConstants.PROXY_SERVICE_TYPE:
+                Map<String, String[]> proxyMap = new HashMap();
+                proxyMap.put(MetricConstants.PROXY_LATENCY_SECONDS, new String[]{serviceName, serviceType});
```

Comment:
Why are we using a map here?

Reply:
Changed the logic to pass the string[] of labels